File: container/omnilint/checkers/json.py
# Copyright (C) 2017 Leandro Lisboa Penz <lpenz@lpenz.org>
# This file is subject to the terms and conditions defined in
# file 'LICENSE', which is part of this source code package.
'''json checker'''

import json

from omnilint.error import Error
from omnilint.checkers import Checker


class Json(Checker):

    extensions = ['json']

    def __init__(self):
        super(Json, self).__init__()

    def check(self, reporter, origname, tmpname, firstline, fd):
        exc = None
        try:
            json.load(fd)
        except json.JSONDecodeError as e:
            exc = e
        if exc is None:
            return
        reporter.report(
            Error(msg=exc.msg,
                  file=origname,
                  line=exc.lineno,
                  column=exc.colno))


def register(omnilint):
    '''Registration function, called by omnilint while loading the checker with
    itself as argument'''
    omnilint.register(Json)

File: tests/json.rs
// Copyright (C) 2026 Leandro Lisboa Penz <lpenz@lpenz.org>
// This file is subject to the terms and conditions defined in
// file 'LICENSE', which is part of this source code package.

//! Integration tests for the analysis of JSON files by the built-in
//! `json-parse` linter, which parses the file contents in-process using the
//! [`serde_json`](https://crates.io/crates/serde_json) crate, together with
//! the external `jq` linter.
//!
//! Requires `jq` to be available on the `PATH`.
//!
//! Only runs when the `test-linter-tools` feature is enabled.

#![cfg(feature = "test-linter-tools")]

mod common;

#[test]
fn clean() {
    assert_eq!(common::run_clean(&["json-clean.json"]), "");
}

#[test]
fn dirty() {
    assert_eq!(
        common::run(&["json-dirty.json"]),
        "Error: lint findings were emitted\n\
         json-dirty.json:1: [jq] Unmatched '}'\n\
         json-dirty.json:1: [json-parse] expected value at line 1 column 15\n"
    );
}
